test/specs/mainnet/exchangeRate/Smoke_ExchangeRate.spec.js

```diff
@@ -0,0 +1,188 @@
+import { EnvNames, NetworkNames, Sdk } from "etherspot";
+import { assert } from "chai";
+import axios from "axios";
+import Helper from "../../../utils/Helper.js";
+
+import * as dotenv from "dotenv";
+dotenv.config(); // init dotenv
+
+let network_etherspot = [
+  "arbitrum",
+  "bsc",
+  "xdai",
+  "matic",
+  "optimism",
+  "mainnet",
+  "fantom",
+  "aurora",
+  "avalanche",
+  "arbitrumNova",
+  "moonbeam",
+  "celo",
+  "fuse",
+];
+
+let network_coinGecho = [
+  "arbitrum-one",
+  "binance-smart-chain",
+  "xdai",
+  "polygon-pos",
+  "optimistic-ethereum",
+  "ethereum",
+  "fantom",
+  "aurora",
+  "avalanche",
+  "arbitrum-nova",
+  "moonbeam",
+  "celo",
+  "fuse",
+];
+
+describe("Compare the Token Rates of the Etherspot and CoinGecho Services.", () => {
+  for (let n = 0; n < network_etherspot.length; n++) {
+    it("Validate the Token Rates of the Etherspot and CoinGecho Services.", async () => {
+      let mainNetSdk;
+      let TOKEN_LIST_ADDRESS_ETHERSPOT = [];
+      let XDAI_CHAIN_ID_ETHERSPOT;
+      let TOKEN_LIST_ADDRESS_COINGECHO = [];
+      let TOKEN_LIST_ID_COINGECHO = [];
+      let responsesCoidList;
+      let requestPayload;
+
+      // initialize the sdk
+      try {
+        mainNetSdk = new Sdk(process.env.PRIVATE_KEY, {
+          env: EnvNames.MainNets,
+          networkName: network_etherspot[n],
+        });
+
+        assert.strictEqual(
+          mainNetSdk.state.accountAddress,
+          "0xa5494Ed2eB09F37b4b0526a8e4789565c226C84f",
+          "The EOA Address is not calculated correctly."
+        );
+      } catch (e) {
+        assert.fail("The SDK is not initialled successfully.");
+      }
+
+      // Compute the smart wallet address
+      try {
+        let smartWalletOutput = await mainNetSdk.computeContractAccount();
+        let smartWalletAddress = smartWalletOutput.address;
+
+        assert.strictEqual(
+          smartWalletAddress,
+          "0x666E17ad27fB620D7519477f3b33d809775d65Fe",
+          "The smart wallet address is not calculated correctly."
+        );
+      } catch (e) {
+        assert.fail("The smart wallet address is not calculated successfully.");
+      }
+
+      // Get the token addresses and it's rate from the CoinGecho
+      try {
+        responsesCoidList = await axios.get(
+          "https://api.coingecko.com/api/v3/coins/list?include_platform=true"
+        );
```

Comment:
Remember to `.catch` and fail should the request / response not function correctly

Reply:
Modified